### ISSUE TITLE
Question about k-center-greedy in PatchCore

### ISSUE THREAD

**Issue by Youskrpig** [openvinotoolkit/anomalib#1332]
Hi，I was wondering why the first randon idx is not included in selected_coreset_idx?
![image](https://github.com/openvinotoolkit/anomalib/assets/51428142/f0010f4c-f1d8-4bbf-a5dc-e86c6a8a67b9)

**Comment by blaz-r** [comment]
Hello. I'm not sure why that is the case, I hope someone more familiar can answer. [Here](https://github.com/amazon-science/patchcore-inspection/blob/main/src/patchcore/sampler.py#L91-L115) is the official implementation of this procedure if it helps. 

Now I'm not sure about all this, but judging from official code and the algorithm from [the Patchcore paper](https://arxiv.org/pdf/2106.08265.pdf)
![image](https://github.com/openvinotoolkit/anomalib/assets/61357777/b5b9b534-049b-4ad9-8ed2-7623f6e937bb) 
I think the implementation in anomalib is correct but a bit different. 

This greedy core-set sub-sampling is an approximation algorithm that "tries to find b center points such that the largest distance between a data point and its nearest center is minimized", from [this paper](https://arxiv.org/pdf/1708.00489.pdf)~[1]. 
![image](https://github.com/openvinotoolkit/anomalib/assets/61357777/a5ea1d45-2521-4479-a95f-df03bf7aed12)

Now in official code what they do to get the initial centers is [take a norm over all distances](https://github.com/amazon-science/patchcore-inspection/blob/main/src/patchcore/sampler.py#L98) and take index of maximum as first feature in core-set. This I think initializes distances to maximum possible value. as it's effectively a sum of all distances from one feature to another, and the feature is just some arbitrary one that is the furthest from all.

In anomalib, instead of this, a random feature is selected, then index of feature with maximum distance from it is stored as the first in core-set, but the feature itself is not added to the core-set. Looking at the paper~[1] that introduced this algorithm, the points at the beginning are chosen uniformly at random and the core-set is returned with those points removed. Which leads me to conclusion that it doesn't really matter how you pick the initial feature and it seems like it doesn't need to be added to core-set.

TLDR (might not be correct): It doesn't matter how the first feature is selected and if it is added to core-set.

**Comment by Youskrpig** [comment]
@blaz-r Thanks for your explanation！
